### PR TITLE
INFRA-620: use centralized actionlint

### PR DESCRIPTION
**Pull request by Kerruba**
## Description

With the PR we move the actionlint job internals into a centralized composite action in order to allow better control over the self-hosted labels used in the repositories

## Checklist

- [x] for CI changes:
  - [x] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

Related to: camunda/team-infrastructure#620

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -59,15 +59,19 @@ jobs:
     timeout-minutes: 5
     env:
       # renovate: datasource=github-releases depName=rhysd/actionlint
-      ACTIONLINT_VERSION: '1.7.4'
+      ACTIONLINT_VERSION: "1.7.4"
```
Comment (cmur2) [inline .github/workflows/ci.yml:62]: This and other changes in this file look like your editor auto-formats the file on save but I'd like to keep this PR to the required changes only, if possible.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -65,10 +65,11 @@ jobs:
       CONFTEST_VERSION: '0.56.0'
     steps:
       - uses: actions/checkout@v4
-      - run: echo "::add-matcher::.github/actionlint-matcher.json"
-      - run: |
-          bash <(curl https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash) $ACTIONLINT_VERSION
-          ./actionlint -shellcheck '' -ignore 'property "vault_.+" is not defined in object type' -ignore 'object type "{}" cannot be filtered by object filtering `.*` since it has no object element'
+      - uses: camunda/infra-global-github-actions/actionlint@main
+        with:
+          ignore: |
```
Comment (cmur2) [inline .github/workflows/ci.yml:70]: Can you pass the `ACTIONLINT_VERSION` defined above?

Comment (Kerruba) [reply .github/workflows/ci.yml:71]: Done ✅

**Review by cmur2 — CHANGES_REQUESTED** [review]
For some reason the new action doesn't behave the same way as executing the `actionlint` binary previously as I tested here:

* [demo PR introducing a problem that actionlint can detect](https://github.com/camunda/camunda/pull/25915) based on `main` gets a failure in the actionlint workflow
* [demo PR introducing the same problem that actionlint can detect](https://github.com/camunda/camunda/pull/25907) based on this PR doesn't not produce any failure or any actionlint output

**Review by cmur2 — APPROVED** [review]

**Pull request merged**